Enable CI

WhyNotHugo opened the pull request:
Enable CI for running tests and deploying.

~Note: PR has been set to use `py3` as a base to simplify diffing, but should actually be merged to master after #1 gets merged.~

Commit:
Run tests via travis

Commit:
Determine package version via setuptools-scm

Commit:
Deploy packages to PyPI via travis

Commit:
Fix an inexisting trove classifier

Commit:
Disable pypy3 for CI

Travis' latest pypy3 version is too old, and fails to install our
dependencies, so just skip it.

Commit:
Document how releasing versions works

Commit:
Use a PyPI-friendly README

PyPI renders RST, so convert the readme to RST when deploying and use
that as a long-description.

TylerHendrickson commented on setup.py:
```diff
@@ -1,94 +1,22 @@
     name='celery-unique',
-    version='1.0.0',
     url='https://github.com/sgrepo/celery-unique',
     author='Tyler Hendrickson, Shiftgig Inc',
```
You should add yourself here and to the next line at some point

Edit: As-discussed, we should move to using a CONTRIBUTORS/AUTHORS file instead

WhyNotHugo replied:
See #3

TylerHendrickson commented on README.rst:
```diff
@@ -1,17 +1,20 @@
+Overview
+--------
 Configuring unique constraints on Celery requires the following:
 - A modified Task base class for Celery (provided by this package)
```
List doesn't appear to be rendering in rst

TylerHendrickson commented on README.rst:
```diff
@@ -1,17 +1,20 @@
-# celery-unique
+celery-unique
```
As-discussed, perhaps we can use `pandoc` to assist with automatically converting `README.md` to `README.rst` as a deployment step.  If that seems feasible, we should change this file back to markdown.

TylerHendrickson commented on .travis.yml:
```diff
@@ -0,0 +1,62 @@
+
+deploy:
+  - provider: pypi
+    user: hobarrera
```
Will provide a Shiftgig user and credentials for you

WhyNotHugo replied:
Ping?

Commit:
Add local eggs to gitignore

Commit:
Remove unused import

hobarrera commented:
I think I screwed up since I hadn't pushed my latest working branch for your last review, and I'd rewritten some commits locally (thought I hadn't pushed them, sorry).

Anyway, this is ready for a second round or review.

Commit:
Use travis-provided credentials for PyPI

Use travis-provided credentials for PyPI, since we want to actually use
shiftgig's account for deploying.
